sdk/textanalytics/Azure.AI.TextAnalytics/src/NamedEntitySubType.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Gets the entity sub-type inferred by the text analytics service's named entity recognition model.
+    /// The list of available types is described at <see href="https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/named-entity-types"/>.
+    /// </summary>
+    [JsonConverter(typeof(NamedEntitySubTypeJsonConverter))]
+    public readonly struct NamedEntitySubType : IEquatable<NamedEntitySubType>
+    {
+        /// <summary>
+        /// Specifies that named entity has no specific sub-type.
+        /// </summary>
+        public static readonly NamedEntitySubType None = default;
```

Comment:
Why not just have this be `NamedEntitySubType("None")` ?

I'm concerned about users having to know they have to call ToString() vs. check for null.  The type is supposed to behave like an enum, and as I understand, the convention in .NET is to have a None value, and there's no concept of a null option in the collection of values.  But please lmk if I'm missing a reason to do it this way instead.